src/test/java/net/smartcosmos/cluster/userdetails/AbstractTestResource.java

```diff
@@ -0,0 +1,67 @@
+package net.smartcosmos.cluster.userdetails;
```

Comment:
There is no reason to create an abstract class when there is no abstract method to implement and there is only one implementation.  This is an unnecessary  class.  Put it all into the test class.  In addition boilerplate code like setup and instance variables are expected in test classes.  You don't want to have hierarchical class dependencies in tests like this, it usually ends up being counter-productive.